doc/features/execution-profiles/README.md

```diff
@@ -0,0 +1,160 @@
+# Execution Profiles
+
+Execution profiles provide a mechanism to group together a set of configuration options and reuse them across different query executions. These options include:
+
+- Load balancing policy
+- Retry policy
+- Speculative Execution policy
+- Consistency level
+- Serial consistency level
+- Per-host request timeout
+
+Execution profiles API is being introduced to help deal with the exploding number of configuration options, especially
+as the database platform evolves into more complex workloads.
+
+The legacy configuration remains intact but it is recommended to set the available options via the new Execution Profiles API.
+
+This page explains how Execution Profiles relate to existing settings, and shows how to use the new profiles for
+request execution.
+
+## Using Execution Profiles
+
+When SLAs are defined there might be different SLAs for different parts of a system. When it comes to authentication, for example, the SLA for a log in might be different from the SLA for a new user registration. Let's say that we need the following settings to meet the SLAs:
+
+| User journey    | Per host timeout  | Consistency |
+|-----------------|-------------------|-------------|
+| Log in          | 100ms             | LOCAL_ONE   |
+| Sign up         | 2000ms            | QUORUM      |
+
+Instead of manually adjusting the options on every request, you can create execution profiles:
+
+```csharp
+var cluster = 
+   Cluster.Builder()
+          .AddContactPoint("127.0.0.1")
+          .WithExecutionProfiles(opts => opts
+            .WithProfile("default", profile => profile
+                .WithLoadBalancingPolicy(new TokenAwarePolicy(new DCAwareRoundRobinPolicy(localDc: "dc1")))
+            .WithProfile("login", profile => profile
+                .WithConsistencyLevel(ConsistencyLevel.LocalOne)
+                .WithReadTimeoutMillis(100)
```

Comment:
I would avoid using read timeout in the example as it's too low level and it shouldn't be used to enforce SLAs (maybe when we introduce "per execution timeout" in the next major, it could make sense).

I think we should point users to spec execs instead.

Reply:
Ok changed the example so that both profiles differ in the `ISpeculativeExecutionPolicy` parameter instead of `ReadTimeoutMillis`.

Reply:
There are still references to read timeout below :)

Reply:
Ah, I really went ham with `ReadTimeoutMillis` on these examples, 😄  fixed